pkg/processor/service.go

```diff
@@ -162,6 +162,7 @@ func (s *Service) Delete(ctx context.Context, id string) error {
 		return err
 	}
 
+	instance.Processor.Close()
```

Comment:
If the delete in the next line fails we will drop inspector instances even though the processor won't be deleted. Again, an edge case, but given that this operation can't fail we might want to execute it after the store delete.

Reply:
Makes sense!